drivers/spi/spi_mcux_lpspi.c

```diff
@@ -170,6 +170,12 @@ static int spi_mcux_configure(struct device *dev,
 		: kLPSPI_MsbFirst;
 
 	master_config.baudRate = spi_cfg->frequency;
+	master_config.betweenTransferDelayInNanoSec =
+		NSEC_PER_SEC / master_config.baudRate * 2;
+	master_config.lastSckToPcsDelayInNanoSec =
+		NSEC_PER_SEC / master_config.baudRate * 2;
+	master_config.pcsToSckDelayInNanoSec =
+		NSEC_PER_SEC / master_config.baudRate * 2;
```

Comment:
I am missing a comment or commit log describing exactly why the delays are calculated like this.

Reply:
The delays calculated refer to hal_nxp https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/drivers/imx/fsl_lpspi.c LPSPI_MasterGetDefaultConfig, Please help check it, Thanks

Reply:
I do understand what the delays are used for. I am asking why you decided to calculate them this way.

Are you calculating them this way based on the timing of a specific device? Is the calculation based on an application note? Are you aiming for a minimum value? If yes, why not simply pass `0` and have the NXP HAL calculate that minimum value?

Currently, I am unable to tell from the code and commit log.

Perhaps it would be better to allow configuring these delays in the device tree? I fear there might be SPI slave devices out there, for which these delays would have to be tweaked.

Reply:
`LPSPI_MasterGetDefaultConfig` calculates the default values of these delays always assuming a default baud rate of 500 kbps. This change reproduces the same calculation from that hal function, but using the requested baud rate.

However, I'm concerned this optimization could inadvertently break existing applications that have been using the default delays, so I think we should wait to merge it until after the 2.3 release.

Reply:
As @MaureenHelm mentioned, HAL NXP only calculates the delay at 500K, not the current SPI rate.
RT1052 RM give the  minimum value SCKPCS = one cycle ,PCSSCK
= one cycle, DBT=2 cycle.
I am not  familiar with SPI of rt1052, so I think reference LPSPI_MasterGetDefaultConfig calculate delay value  is a good way.
![timing](https://user-images.githubusercontent.com/3456246/82731907-8567ea80-9d3c-11ea-8582-3762c1447a8b.png)

Reply:
@henrikbrixandersen and I discussed this in the release readiness meeting yesterday (his platforms are most affected by this change), and we agreed it's ok to fix this in 2.3 as long as the delays are made configurable through optional device tree properties. If a node doesn't an define optional property, then the driver defaults to this calculation. Can you update your PR to do this in the next couple of days, so it can be included in the release?